File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/maps/Annotations.java
package org.maplibre.android.maps;


import androidx.annotation.NonNull;

import org.maplibre.android.annotations.Annotation;

import java.util.List;

/**
 * Interface that defines convenient methods for working with a {@link Annotation}'s collection.
 */
interface Annotations {
  Annotation obtainBy(long id);

  List<Annotation> obtainAll();

  void removeBy(long id);

  void removeBy(@NonNull Annotation annotation);

  void removeBy(@NonNull List<? extends Annotation> annotationList);

  void removeAll();
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/Annotations.kt
package org.maplibre.android.maps

import org.maplibre.android.annotations.Annotation

/**
 * Interface that defines convenient methods for working with a [Annotation]'s collection.
 */
internal interface Annotations {
    fun obtainBy(id: Long): Annotation?

    fun obtainAll(): List<Annotation>

    fun removeBy(id: Long)

    fun removeBy(annotation: Annotation)

    fun removeBy(annotationList: List<Annotation>)

    fun removeAll()
}
